IOMC/Input/plugins/HepMCEventWriter.cc

```diff
@@ -47,8 +45,7 @@ void HepMCEventWriter::endRun(const edm::Run &run, const edm::EventSetup &es) {
 }
 
 void HepMCEventWriter::analyze(const edm::Event &event, const edm::EventSetup &es) {
-  edm::Handle<edm::HepMCProduct> product;
-  event.getByLabel(hepMCProduct_, product);
+  const edm::Handle<edm::HepMCProduct> &product = event.getHandle(hepMCProduct_);
```

Comment:
So this code has not been functional for ~8 years, and no-one has complained. I think it would be safe to remove the class and the only configuration file that uses it
https://github.com/cms-sw/cmssw/blob/3b039e0e0c9cbf71bd9e1f840339a1c10c82dfd5/IOMC/Input/test/testWriter_cfg.py#L117

@cms-sw/generators-l2 This EDModule seems to be more in your purview, do you see any need for this kind of EDAnalyzer?

Reply:
I interpret the silence as this code is not needed. @bsunanda, could you update the PR (or make a separate one) removing the class and `IOMC/Input/test/testWriter_cfg.py` configuration file?

Reply:
@makortel in the far past, at the beginning of data taking, we used generated events in ascii file format to share in a simple way exactly the same set of events between different collaborations and make comparisons of simulation performances exactly on the same input. I am not sure whether this has any more used since then and whether it could have some usefulness today, @civanch and @bsunanda might better comment about this.

Reply:
Thanks @fabiocos for your comment for the history. I'd still be inclined to make a prediction based on past, and clean it up now. If a use case arises later, the code can always be resurrected.